### PR TITLE
Feat 89

### PR DESCRIPTION
**Pull request by StachoB**
We, Barbara STACHOWICZ, LIliane LI, Ziwei GONG, Shengjie ZHOU, Yichen LIU hereby grant to Hyperglosae maintainers the right to publish our contribution under the terms of any licenses the Free Software Foundation classifies as Free Software Licenses.

### CODE REVIEW

File: frontend/src/Page.js [hunk 1]
```diff
@@ -21,6 +22,7 @@ function Page({backend}) {
   const [scholiaMetadata, setScholiaMetadata] = useState([]);
   const [content, setContent] = useState([]);
   const [lastUpdate, setLastUpdate] = useState();
+  const [isCollection, setIsCollection] = useState(true);
```
Comment (benel) [inline frontend/src/Page.js:25]: Do you really a need a state for this?
What does it mean exactly?

File: frontend/src/Page.js [hunk 2]
```diff
@@ -60,8 +62,11 @@ function Page({backend}) {
         x => !forwardLinks.includes(x._id) && x._id !== id
       );
       setScholiaMetadata(reverseLinkedDocuments);
+      console.log(forwardLinks.length);
+      forwardLinks.length ? setIsCollection(true) : setIsCollection(false);
+      console.log(isCollection);
```
Comment (benel) [inline frontend/src/Page.js:67]: Please don't add debug logs to your commits.

File: frontend/src/Page.js [hunk 3]
```diff
@@ -60,8 +62,11 @@ function Page({backend}) {
         x => !forwardLinks.includes(x._id) && x._id !== id
       );
       setScholiaMetadata(reverseLinkedDocuments);
+      console.log(forwardLinks.length);
```
Comment (benel) [inline frontend/src/Page.js:65]: Please don't add debug logs to your commits.

File: frontend/src/Page.js [hunk 4]
```diff
@@ -104,7 +109,7 @@ function Page({backend}) {
     <Container className="screen">
       <Row>
         <Col md={2} className="sources">
-          <DocumentsCards docs={sourcesOfSourceMetadata} byRow={1} />
+          <DocumentsCards docs={sourcesOfSourceMetadata} byRow={1}/>
```
Comment (benel) [inline frontend/src/Page.js:112]: Please don't add lines or files unrelated with your contribution.
This makes integration harder (with more conflicts) and git blame less reliable.

**Review by benel — CHANGES_REQUESTED** [review]
Thank you very much for your contribution @Hypertopic/arts-plastiques.

- Please fix the problems spotted in my code review.
- Please apply also the rules for commits (coauthors ; exactly 3 commits: SCENARIO, TEST, FEATURE ; `#` with numbers, `closes` instead of `see` for the FEATURE, period at the end).

File: frontend/package-lock.json [hunk 5]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "frontend",
   "version": "0.1.0",
-  "lockfileVersion": 2,
+  "lockfileVersion": 3,
```
Comment (benel) [inline frontend/package-lock.json:4]: Please don't edit lines or files unrelated with your contribution.

File: frontend/src/DocumentsCards.js [hunk 6]
```diff
@@ -5,8 +5,9 @@ import Row from 'react-bootstrap/Row';
 import Metadata from './Metadata';
 import BrowseTools from './BrowseTools';
 import FutureDocument from './FutureDocument';
+import Creation from './Creation';
 
-function DocumentsCards({docs, expandable, byRow, createOn, setLastUpdate, backend}) {
+function DocumentsCards({docs, expandable, byRow, createOn, setLastUpdate, backend, isCollection}) {
```
[the remark this answers is not in the document]
Comment (ShengjieZhou) [reply frontend/src/DocumentsCards.js:10]: Here we check if the document is the collection before showing the button to create collection from the image, this is why we use isCollection.

File: frontend/src/Page.js (hunk 1 above)
Comment (ShengjieZhou) [reply frontend/src/Page.js:25]: We have tried the simple variable, but it not works so that we have to use useState.

**Comment by benel** [comment]
Pour information, vos scénarios ne tenaient pas compte de l'authentification (ajoutée le 1er mai dernier).
Voir https://github.com/Hypertopic/HyperGlosae/commit/e9fd5c4180f7dd0d267fb8a0887ffac9185c3984#commitcomment-111471471.

File: frontend/src/DocumentsCards.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (benel) [reply frontend/src/DocumentsCards.js:10]: > Here we check if the document is the collection before showing the button to create collection from the image, this is why we use isCollection.

@ShengjieZhou @lililiane @Ziwei-GONG-UTT 

This seems confusing to me...
- If the focused document is not a collection, you prevent users to create a collection from it.
- If the document is a collection, you allow users to create a collection from it.

What is the purpose of this?

**Pull request closed without merging**